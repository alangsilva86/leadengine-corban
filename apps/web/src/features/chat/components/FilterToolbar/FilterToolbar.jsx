import { useMemo, useState } from 'react';
import { Button } from '@/components/ui/button.jsx';
import { Input } from '@/components/ui/input.jsx';
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from '@/components/ui/select.jsx';
import { Popover, PopoverContent, PopoverTrigger } from '@/components/ui/popover.jsx';
import { cn } from '@/lib/utils.js';
import { Filter, Loader2, MessageSquarePlus, RefreshCw, Search, X } from 'lucide-react';

const DEFAULT_FILTERS = {
  scope: 'team',
  state: 'open',
  window: 'in_window',
  outcome: 'all',
};

const SCOPE_OPTIONS = [
  { value: 'team', label: 'Equipe' },
  { value: 'mine', label: 'Meus' },
];

const STATE_OPTIONS = [
  { value: 'open', label: 'Abertos' },
  { value: 'pending', label: 'Pendentes' },
  { value: 'assigned', label: 'Atribuídos' },
  { value: 'resolved', label: 'Resolvidos' },
];

const WINDOW_OPTIONS = [
  { value: 'in_window', label: 'Janela 24h' },
  { value: 'expired', label: 'Expirados' },
];

const OUTCOME_OPTIONS = [
  { value: 'all', label: 'Todos' },
  { value: 'won', label: 'Ganhos' },
  { value: 'lost', label: 'Perdidos' },
];

const FilterToolbar = ({
  search,
  onSearchChange,
  filters,
  onFiltersChange,
  loading,
  onRefresh,
  onStartManualConversation,
  manualConversationPending = false,
}) => {
  const [filtersOpen, setFiltersOpen] = useState(false);

  const effectiveFilters = useMemo(
    () => ({
      scope: filters?.scope ?? DEFAULT_FILTERS.scope,
      state: filters?.state ?? DEFAULT_FILTERS.state,
      window: filters?.window ?? DEFAULT_FILTERS.window,
      outcome: filters?.outcome ?? DEFAULT_FILTERS.outcome,
    }),
    [filters]
  );

  const activeFiltersCount = useMemo(
    () =>
      Object.entries(effectiveFilters).reduce(
        (count, [key, value]) => (DEFAULT_FILTERS[key] !== value ? count + 1 : count),
        0
      ),
    [effectiveFilters]
  );

  const applyFilters = (updater) => {
    if (typeof onFiltersChange !== 'function') return;
    onFiltersChange((current) => {
      const next = typeof updater === 'function' ? updater(current) : updater;
      return { ...current, ...next };
    });
  };

  const getOptionLabel = (options, value) => options.find((option) => option.value === value)?.label ?? value;

  const filterSummaries = useMemo(
    () => [
      { id: 'scope', label: 'Responsável', value: getOptionLabel(SCOPE_OPTIONS, effectiveFilters.scope) },
      { id: 'state', label: 'Status', value: getOptionLabel(STATE_OPTIONS, effectiveFilters.state) },
      { id: 'window', label: 'Janela', value: getOptionLabel(WINDOW_OPTIONS, effectiveFilters.window) },
      { id: 'outcome', label: 'Resultado', value: getOptionLabel(OUTCOME_OPTIONS, effectiveFilters.outcome) },
    ],
    [effectiveFilters]
  );

  const activeFilterSummaries = useMemo(
    () =>
      filterSummaries.filter((summary) => effectiveFilters[summary.id] !== DEFAULT_FILTERS[summary.id]),
    [effectiveFilters, filterSummaries]
  );

  const handleResetFilters = () => {
    applyFilters(DEFAULT_FILTERS);
    setFiltersOpen(false);
  };

  const handleRemoveFilter = (filterId) => {
    if (!(filterId in DEFAULT_FILTERS)) return;
    applyFilters({ [filterId]: DEFAULT_FILTERS[filterId] });
  };

  return (
    <div className="flex w-full flex-col gap-2 text-[color:var(--color-inbox-foreground)]">
      <div className="flex flex-wrap items-center gap-2 sm:gap-3">
        <div className="relative flex min-w-[200px] flex-1 items-center sm:max-w-md">
          <Search className="pointer-events-none absolute left-3 top-1/2 h-4 w-4 -translate-y-1/2 text-[color:var(--color-inbox-foreground-muted)]" />
          <Input
            value={search}
            onChange={(event) => onSearchChange?.(event.target.value)}
            placeholder="Buscar tickets, contatos..."
            className="h-9 w-full rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] pl-9 text-sm text-[color:var(--color-inbox-foreground)] placeholder:text-[color:var(--color-inbox-foreground-muted)]"
          />
        </div>

        <Popover open={filtersOpen} onOpenChange={setFiltersOpen}>
          <PopoverTrigger asChild>
            <Button
              variant="outline"
              size="sm"
              className="h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] px-3 text-xs font-medium text-[color:var(--color-inbox-foreground)] hover:bg-[color:color-mix(in_srgb,var(--surface-overlay-inbox-bold)_92%,transparent)]"
            >
              <Filter className="h-4 w-4" />
              <span className="ml-2 hidden sm:inline">Filtros</span>
              {activeFiltersCount > 0 ? (
                <span className="ml-2 inline-flex h-5 min-w-[1.5rem] items-center justify-center rounded-full bg-[color:color-mix(in_srgb,var(--accent-inbox-primary)_18%,transparent)] px-2 text-[10px] font-semibold uppercase tracking-wide text-[color:var(--accent-inbox-primary)]">
                  {activeFiltersCount}
                </span>
              ) : null}
            </Button>
          </PopoverTrigger>
          <PopoverContent
            align="end"
            className="w-[320px] rounded-xl border border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] p-4 text-[color:var(--color-inbox-foreground)] shadow-[var(--shadow-lg)]"
          >
            <div className="space-y-4">
              <div className="space-y-2">
                <p className="text-xs font-semibold uppercase tracking-wide text-[color:var(--color-inbox-foreground-muted)]">Responsável</p>
                <Select
                  value={effectiveFilters.scope}
                  onValueChange={(value) => applyFilters({ scope: value })}
                >
                  <SelectTrigger className="h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] text-sm text-[color:var(--color-inbox-foreground)]">
                    <SelectValue placeholder="Equipe" />
                  </SelectTrigger>
                  <SelectContent className="border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-quiet)] text-[color:var(--color-inbox-foreground)]">
                    {SCOPE_OPTIONS.map((option) => (
                      <SelectItem key={option.value} value={option.value}>
                        {option.label}
                      </SelectItem>
                    ))}
                  </SelectContent>
                </Select>
              </div>

              <div className="space-y-2">
                <p className="text-xs font-semibold uppercase tracking-wide text-[color:var(--color-inbox-foreground-muted)]">Status</p>
                <Select
                  value={effectiveFilters.state}
                  onValueChange={(value) => applyFilters({ state: value })}
                >
                  <SelectTrigger className="h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] text-sm text-[color:var(--color-inbox-foreground)]">
                    <SelectValue placeholder="Status" />
                  </SelectTrigger>
                  <SelectContent className="border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-quiet)] text-[color:var(--color-inbox-foreground)]">
                    {STATE_OPTIONS.map((option) => (
                      <SelectItem key={option.value} value={option.value}>
                        {option.label}
                      </SelectItem>
                    ))}
                  </SelectContent>
                </Select>
              </div>

              <div className="space-y-2">
                <p className="text-xs font-semibold uppercase tracking-wide text-[color:var(--color-inbox-foreground-muted)]">Janela</p>
                <Select
                  value={effectiveFilters.window}
                  onValueChange={(value) => applyFilters({ window: value })}
                >
                  <SelectTrigger className="h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] text-sm text-[color:var(--color-inbox-foreground)]">
                    <SelectValue placeholder="Janela" />
                  </SelectTrigger>
                  <SelectContent className="border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-quiet)] text-[color:var(--color-inbox-foreground)]">
                    {WINDOW_OPTIONS.map((option) => (
                      <SelectItem key={option.value} value={option.value}>
                        {option.label}
                      </SelectItem>
                    ))}
                  </SelectContent>
                </Select>
              </div>

              <div className="space-y-2">
                <p className="text-xs font-semibold uppercase tracking-wide text-[color:var(--color-inbox-foreground-muted)]">Resultado</p>
                <Select
                  value={effectiveFilters.outcome}
                  onValueChange={(value) => applyFilters({ outcome: value })}
                >
                  <SelectTrigger className="h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] text-sm text-[color:var(--color-inbox-foreground)]">
                    <SelectValue placeholder="Resultado" />
                  </SelectTrigger>
                  <SelectContent className="border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-quiet)] text-[color:var(--color-inbox-foreground)]">
                    {OUTCOME_OPTIONS.map((option) => (
                      <SelectItem key={option.value} value={option.value}>
                        {option.label}
                      </SelectItem>
                    ))}
                  </SelectContent>
                </Select>
              </div>

              <div className="flex items-center justify-between">
                <Button
                  variant="ghost"
                  size="sm"
                  className="h-8 rounded-lg px-3 text-xs font-medium text-[color:var(--color-inbox-foreground-muted)] hover:text-[color:var(--color-inbox-foreground)]"
                  onClick={handleResetFilters}
                >
                  Limpar filtros
                </Button>
                <Button
                  size="sm"
                  className="h-8 rounded-lg bg-[color:var(--accent-inbox-primary)] px-4 text-xs font-semibold text-white hover:bg-[color:color-mix(in_srgb,var(--accent-inbox-primary)_92%,transparent)]"
                  onClick={() => setFiltersOpen(false)}
                >
                  Aplicar
                </Button>
              </div>
            </div>
          </PopoverContent>
        </Popover>

        <Button
          variant="outline"
          size="sm"
          className="h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] px-3 text-xs font-medium text-[color:var(--color-inbox-foreground)] hover:bg-[color:color-mix(in_srgb,var(--surface-overlay-inbox-bold)_92%,transparent)]"
          onClick={onRefresh}
          disabled={loading}
        >
          <RefreshCw className={cn('h-4 w-4', loading && 'animate-spin')} />
          <span className="ml-2 hidden sm:inline">Atualizar</span>
        </Button>

        {onStartManualConversation ? (
          <Button
            size="sm"
            className="h-9 rounded-lg bg-success px-3 text-xs font-semibold text-success-foreground hover:bg-success/90 disabled:opacity-70"
            onClick={onStartManualConversation}
            disabled={manualConversationPending}
          >
            {manualConversationPending ? (
              <Loader2 className="h-4 w-4 animate-spin" />
            ) : (
              <MessageSquarePlus className="h-4 w-4" />
            )}
            <span className="ml-2 hidden sm:inline">Nova conversa</span>
          </Button>
        ) : null}
      </div>

      {activeFilterSummaries.length ? (
        <div className="flex flex-wrap items-center gap-2 text-xs text-[color:var(--color-inbox-foreground-muted)]">
          {activeFilterSummaries.map((summary) => (
            <button
              key={summary.id}
              type="button"
              className="inline-flex items-center gap-2 rounded-full border border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] px-3 py-1 text-xs font-medium text-[color:var(--color-inbox-foreground)] transition hover:bg-[color:color-mix(in_srgb,var(--surface-overlay-inbox-bold)_92%,transparent)] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:var(--accent-inbox-primary)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:var(--surface-shell)]"
              onClick={() => handleRemoveFilter(summary.id)}
            >
              <span className="text-[color:var(--color-inbox-foreground-muted)]">{summary.label}:</span>
              <span>{summary.value}</span>
              <X className="h-3 w-3" />
            </button>
          ))}
          <button
            type="button"
            className="inline-flex items-center gap-1 text-xs font-semibold uppercase tracking-wide text-[color:var(--color-inbox-foreground-muted)] hover:text-[color:var(--color-inbox-foreground)]"
            onClick={handleResetFilters}
          >
            Limpar tudo
            <X className="h-3 w-3" />
          </button>
        </div>
      ) : null}
    </div>
  );
};

export default FilterToolbar;
