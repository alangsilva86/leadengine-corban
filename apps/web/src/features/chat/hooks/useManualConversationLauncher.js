import { useMutation, useQueryClient } from '@tanstack/react-query';
import { apiPost } from '@/lib/api.js';
import { looksLikeWhatsAppJid } from '@/features/whatsapp/utils/instanceIdentifiers.js';

const sanitizePhone = (value) => String(value ?? '').replace(/\D/g, '');

const buildChatId = (value) => {
  if (!value) {
    return null;
  }

  if (looksLikeWhatsAppJid(value)) {
    return value;
  }

  const digits = sanitizePhone(value);
  if (!digits) {
    return null;
  }

  return digits;
};

export const useManualConversationLauncher = () => {
  const queryClient = useQueryClient();

  const mutation = useMutation({
    mutationKey: ['lead-inbox', 'manual-conversation'],
    mutationFn: async ({ phone, message, instanceId }) => {
      const chatId = buildChatId(phone);
      const trimmedMessage = typeof message === 'string' ? message.trim() : '';
      const normalizedInstance = typeof instanceId === 'string' ? instanceId.trim() : '';

      if (!chatId) {
        throw new Error('Informe um telefone válido com DDD e país.');
      }

      if (!trimmedMessage) {
        throw new Error('Digite a mensagem inicial.');
      }

      if (!normalizedInstance) {
        throw new Error('Selecione uma instância conectada.');
      }

      const response = await apiPost('/api/tickets/messages', {
        chatId,
        iid: normalizedInstance,
        text: trimmedMessage,
        metadata: {
          origin: 'manual-conversation',
          phone: sanitizePhone(phone),
        },
      });

      const data = response?.data ?? response ?? null;

      return {
        ticket: data?.ticket ?? null,
        ticketId:
          data?.ticketId ??
          data?.ticket?.id ??
          data?.message?.ticketId ??
          null,
        message: data?.message ?? null,
        raw: data,
      };
    },
    onSuccess: (result) => {
      if (result?.ticketId) {
        queryClient.invalidateQueries({
          queryKey: ['chat', 'messages', result.ticketId],
        });
      }
      queryClient.invalidateQueries({ queryKey: ['chat', 'tickets'] });
    },
  });

  return {
    launch: (payload) => mutation.mutateAsync(payload),
    isPending: mutation.isPending,
    error: mutation.error,
    data: mutation.data,
    reset: mutation.reset,
    isAvailable: true,
    unavailableReason: null,
  };
};

export default useManualConversationLauncher;
