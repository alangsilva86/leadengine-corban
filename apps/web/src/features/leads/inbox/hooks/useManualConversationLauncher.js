import { useMutation } from '@tanstack/react-query';

import { apiGet, apiPost } from '@/lib/api.js';

const sanitizePhone = (value) => String(value ?? '').replace(/\D/g, '');

const fetchContactByPhone = async (digits) => {
  if (!digits) {
    return null;
  }

  const response = await apiGet(`/api/contacts?phone=${encodeURIComponent(digits)}`);
  const items = response?.data?.items;
  if (Array.isArray(items) && items.length > 0) {
    return items[0];
  }

  return null;
};

const createContact = async (digits) => {
  const defaultName = `Contato ${digits}`;
  const response = await apiPost('/api/contacts', {
    name: defaultName,
    phone: digits,
  });
  return response?.data ?? null;
};

export const useManualConversationLauncher = () => {
  const mutation = useMutation({
    mutationKey: ['lead-inbox', 'manual-conversation'],
    mutationFn: async ({ phone, message }) => {
      const digits = sanitizePhone(phone);
      const trimmedMessage = typeof message === 'string' ? message.trim() : '';

      if (!digits) {
        throw new Error('Informe um telefone válido.');
      }

      let contact = await fetchContactByPhone(digits);

      if (!contact) {
        try {
          contact = await createContact(digits);
        } catch (error) {
          const status = error?.status ?? error?.statusCode;
          if (status === 409) {
            contact = await fetchContactByPhone(digits);
          } else {
            throw error;
          }
        }
      }

      if (!contact) {
        throw new Error('Não foi possível localizar ou criar o contato.');
      }

      const leadResponse = await apiPost('/api/leads', {
        contactId: contact.id,
        source: 'WHATSAPP',
        notes: trimmedMessage || undefined,
      });

      const lead = leadResponse?.data ?? null;

      return {
        contact,
        lead,
        phone: digits,
        message: trimmedMessage,
      };
    },
  });

  return {
    launch: (payload) => mutation.mutateAsync(payload),
    isPending: mutation.isPending,
    error: mutation.error,
    data: mutation.data,
    reset: mutation.reset,
  };
};

export default useManualConversationLauncher;
