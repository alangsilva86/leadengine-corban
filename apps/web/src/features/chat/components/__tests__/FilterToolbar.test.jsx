/** @vitest-environment jsdom */
import '@testing-library/jest-dom/vitest';
import { cleanup, render, screen } from '@testing-library/react';
import { afterEach, describe, expect, it } from 'vitest';

import { MANUAL_CONVERSATION_DEPRECATION_MESSAGE } from '../../hooks/useManualConversationLauncher.js';
import FilterToolbar from '../FilterToolbar/FilterToolbar.jsx';

describe('FilterToolbar', () => {
  afterEach(() => {
    cleanup();
  });

  const defaultProps = {
    search: '',
    onSearchChange: () => {},
    filters: {},
    onFiltersChange: () => {},
    loading: false,
    onRefresh: () => {},
  };

  it('omite a ação de nova conversa e exibe aviso quando o fluxo manual está indisponível', () => {
    const { container } = render(
      <FilterToolbar
        {...defaultProps}
        manualConversationUnavailableReason={MANUAL_CONVERSATION_DEPRECATION_MESSAGE}
      />
    );

    expect(screen.queryByRole('button', { name: /nova conversa/i })).not.toBeInTheDocument();
    expect(screen.getByText(MANUAL_CONVERSATION_DEPRECATION_MESSAGE)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
