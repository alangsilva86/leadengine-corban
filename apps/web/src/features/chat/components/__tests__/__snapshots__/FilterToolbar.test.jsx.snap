// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FilterToolbar > omite a ação de nova conversa e exibe aviso quando o fluxo manual está indisponível 1`] = `
<div>
  <div
    class="flex w-full flex-col gap-2 text-[color:var(--color-inbox-foreground)]"
  >
    <div
      class="flex flex-wrap items-center gap-2 sm:gap-3"
    >
      <div
        class="relative flex min-w-[200px] flex-1 items-center sm:max-w-md"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-search pointer-events-none absolute left-3 top-1/2 h-4 w-4 -translate-y-1/2 text-[color:var(--color-inbox-foreground-muted)]"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21 21-4.34-4.34"
          />
          <circle
            cx="11"
            cy="11"
            r="8"
          />
        </svg>
        <input
          class="file:text-foreground selection:bg-primary selection:text-primary-foreground dark:bg-[color-mix(in_srgb,var(--input)_18%,transparent)] flex min-w-0 border px-3 py-1 shadow-[0_0_0_1px_color-mix(in_srgb,var(--border)_70%,transparent)] transition-[color,box-shadow,border-color] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-transparent focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] focus-visible:shadow-[0_0_0_1px_color-mix(in_srgb,var(--ring)_75%,transparent)] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive h-9 w-full rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] pl-9 text-sm text-[color:var(--color-inbox-foreground)] placeholder:text-[color:var(--color-inbox-foreground-muted)]"
          data-slot="input"
          placeholder="Buscar tickets, contatos..."
          value=""
        />
      </div>
      <button
        aria-controls="radix-_r_0_"
        aria-expanded="false"
        aria-haspopup="dialog"
        class="inline-flex items-center justify-center whitespace-nowrap transition-all disabled:pointer-events-none disabled:opacity-60 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] border shadow-[0_0_0_1px_color-mix(in_srgb,var(--border)_55%,transparent)] min-h-[38px] gap-1.5 has-[>svg]:px-2.5 h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] px-3 text-xs font-medium text-[color:var(--color-inbox-foreground)] hover:bg-[color:color-mix(in_srgb,var(--surface-overlay-inbox-bold)_92%,transparent)]"
        data-slot="popover-trigger"
        data-state="closed"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-funnel h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10 20a1 1 0 0 0 .553.895l2 1A1 1 0 0 0 14 21v-7a2 2 0 0 1 .517-1.341L21.74 4.67A1 1 0 0 0 21 3H3a1 1 0 0 0-.742 1.67l7.225 7.989A2 2 0 0 1 10 14z"
          />
        </svg>
        <span
          class="ml-2 hidden sm:inline"
        >
          Filtros
        </span>
      </button>
      <button
        class="inline-flex items-center justify-center whitespace-nowrap transition-all disabled:pointer-events-none disabled:opacity-60 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[color:color-mix(in_srgb,var(--ring)_75%,transparent)] focus-visible:ring-offset-2 focus-visible:ring-offset-[color:color-mix(in_srgb,var(--bg)_92%,transparent)] border shadow-[0_0_0_1px_color-mix(in_srgb,var(--border)_55%,transparent)] min-h-[38px] gap-1.5 has-[>svg]:px-2.5 h-9 rounded-lg border-[color:var(--color-inbox-border)] bg-[color:var(--surface-overlay-inbox-bold)] px-3 text-xs font-medium text-[color:var(--color-inbox-foreground)] hover:bg-[color:color-mix(in_srgb,var(--surface-overlay-inbox-bold)_92%,transparent)]"
        data-slot="button"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-refresh-cw h-4 w-4"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 12a9 9 0 0 1 9-9 9.75 9.75 0 0 1 6.74 2.74L21 8"
          />
          <path
            d="M21 3v5h-5"
          />
          <path
            d="M21 12a9 9 0 0 1-9 9 9.75 9.75 0 0 1-6.74-2.74L3 16"
          />
          <path
            d="M8 16H3v5"
          />
        </svg>
        <span
          class="ml-2 hidden sm:inline"
        >
          Atualizar
        </span>
      </button>
    </div>
    <div
      class="flex items-start gap-3 rounded-lg border border-dashed border-[color:var(--color-inbox-border)] bg-[color:color-mix(in_srgb,var(--surface-overlay-inbox-bold)_80%,transparent)] px-4 py-3 text-xs text-[color:var(--color-inbox-foreground)]"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-triangle-alert mt-0.5 h-4 w-4 flex-none text-[color:var(--accent-inbox-primary)]"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m21.73 18-8-14a2 2 0 0 0-3.48 0l-8 14A2 2 0 0 0 4 21h16a2 2 0 0 0 1.73-3"
        />
        <path
          d="M12 9v4"
        />
        <path
          d="M12 17h.01"
        />
      </svg>
      <div
        class="flex-1 leading-relaxed text-[color:var(--color-inbox-foreground-muted)]"
      >
        O fluxo manual de novas conversas foi aposentado. Utilize o canal oficial de abertura de tickets.
      </div>
    </div>
  </div>
</div>
`;
